### PR TITLE
Adds injectable prophylactic antivenom

### PR DESCRIPTION
**Pull request by aoae23**
#### Summary

SUMMARY: Content "Adds prophylactic antivenom"

#### Purpose of change

Adds an interesting (at least in my opinion) high-tech drug that allows the player to temporarily resist venom when taken in advance. It is thematic in the context of the lore, developed in response to the threat of accelerated chemical and biological weapons development in rival nations. 
They serve a much-needed way to mitigate the threat of insect venoms, which are significantly less trivial thanks to the work of @Venera3. However, the spawn location of these pills (only military locations) should make them difficult to obtain early on.

#### Describe the solution

Adds them

#### Describe alternatives you've considered

Using the spell infrastructure to add the effect, but I haven't looked into the documentation for it yet.

#### Testing

1. Ensure that it spawns in the correct locations
2. Ensure that the game doesn't break when they are taken
3. Ensure that the effect lasts for some time when taken, and properly resists venoms from giant wasps, etc.
4. Ensure that the effect properly dissipates after a week

#### Additional context

~~I don't know how to code in C++ - most of this was hacked together from existing iuse code~~ fortunately this can be done entirely in json

### CODE REVIEW

File: data/json/items/comestibles/med.json [hunk 1]
```diff
@@ -1402,6 +1402,24 @@
...
+    "weight": "1 g",
+    "volume": "250 ml",
+    "price": 10000,
+    "price_postapoc": 50000,
+    "charges": 20,
+    "stack_size": 200,
+    "symbol": "!",
+    "color": "white",
+    "healthy": -1,
+    "use_action": [ "PROPHYLACTIC_ANTIVENOM" ],
```
Comment (anothersimulacrum) [inline data/json/items/comestibles/med.json:1420]: You can use the consume_drug use action, scroll up to poppy painkiller to see an example. Duration can be specified as a string, as well.

Comment (aoae23) [reply data/json/items/comestibles/med.json]: In that case, should I avoid fiddling with the effects in item_factory.cpp and item_actions.json?

Comment (anothersimulacrum) [reply data/json/items/comestibles/med.json]: Yep, no C++ changes required then.

File: src/iuse.cpp [hunk 2]
```diff
@@ -939,6 +940,14 @@ int iuse::flu_vaccine( player *p, item *it, bool, const tripoint & )
     return it->type->charges_to_use();
 }
 
+int iuse::prophylactic_antivenom( player *p, item *it, bool, const tripoint & )
+{
+    p->add_msg_if_player( m_good,
+                          _( "You feel resistant to venoms." ) );
+    p->add_effect( effect_took_prophylactic_antivenom, 1_weeks, true );
```
Comment (actual-nh) [inline src/iuse.cpp:947]:
```suggestion
    p->add_effect( effect_prophylactic_antivenom, 1_weeks, true );
```
(As you have almost certainly also noticed...)

File: data/json/items/comestibles/med.json [hunk 3]
```diff
@@ -1402,6 +1402,28 @@
...
+    "charges": 20,
+    "stack_size": 200,
+    "symbol": "!",
+    "color": "white",
+    "healthy": -1,
+    "use_action": {
+      "type": "consume_drug",
+      "activation_message": "You take some prophylactic antivenom pills.",
+      "effects": [ { "id": "prophylactic_antivenom", "duration": "1 w" } ]
+    }
```
Comment (kevingranade) [inline data/json/items/comestibles/med.json:1424]:
```suggestion
    },
```

File: src/iuse.cpp [hunk 4]
```diff
@@ -192,6 +192,7 @@ static const efftype_id effect_music( "music" );
 static const efftype_id effect_onfire( "onfire" );
 static const efftype_id effect_paincysts( "paincysts" );
 static const efftype_id effect_pet( "pet" );
+static const efftype_id effect_prophylactic_antivenom ( "prophylactic_antivenom" );
```
Comment (kevingranade) [inline src/iuse.cpp:195]:
```suggestion
static const efftype_id effect_prophylactic_antivenom( "prophylactic_antivenom" );
```

**Comment by Venera3** [comment]
I did leave in the old venom effects ("poison" and "badpoison") as well, so you might want to add them too for consistency's sake.

**Comment by aoae23** [comment]
resist_effects makes much more sense here, thanks for the suggestion.

**Comment by kevingranade** [comment]
I like the general idea of adding antivenoms.
Balance-wise, this needs to be reigned in a good bit.  There are a number of properties that tweak effectiveness, and we need to exercise a number of them. Right now, taking a dose of this stuff immediately upon finding it and keeping whatever additional doses you find around is a complete no-brainer, which isn't what we want.  It also removes the sole special ability from a whole type of enemy.

Specificity: This is broad-spectrum, which increases the effective utility of the item a great deal. If there were several versions of antivenom, each of which only applied to one venom type, that would reduce the effective utility. That would also help with representativeness, as targeted antivenoms exist IRL. It would also allow differentiation between different antivenom types, some would be relative common, for example ones that exist for pre-cataclysm creatures like spiders and snakes.
Portability: As a pill, this is extremely high. There is essentially zero opportunity cost for carrying some around. If it were a compound that required refrigeration or even just special handling (e.g. represented by a larger padded case you're required to carry a vial of the drug in), that would reduce effective utility.
Duration: Currently it's extremely long, practically unlimited to the point where finding a cache of the stuff means you're covered for the rest of the game (12 doses = 2 months, plenty of time to find more).  It would be a lot more interesting if the duration was more similar to that of venoms themselves, essentially requiring you to take one before combat with an envenoming creature rather than granting a durable resistance.  The number of doses makes this worse, if instead of a pill bottle with several months of doses, it were an injectible vial with a single dose, that would at least make people consider holding one in reserve.
Effectiveness: AFAIK this makes you fully immune, a chance to resist increases in venom intensity or similar would be a significant reduction in effectiveness, but also requires more infrastructure. Even a 90% chance to resist would greatly impact the balance, since you're no longer completely immune.
Side effects: A down side to such a potent treatment would also make things more interesting, even a marginal but durable penalty like a single-digit speed penalty, or an endurance or eariness penalty would make people think twice before taking the drug without a thought.

What I recommend as a balance between ease of implementation and balance and making it interesting is to change it to:
An autoinjector (for bulk) that has a single dose. Think like an epi-pen.
Triggering it causes minor pain and something minor like nausea.
Duration is several hours.
It can stay broad-spectrum.

If you want different versions with different venom targets, that works too, and you can explore giving them slightly different tradeoffs, and some of them could be more common.

**Comment by Venera3** [comment]
Numbers-wise assuming max intensity, you're looking at ~10 dmg and max 20 pain for the damaging venom ( down from ~40 and 120, respectively), and -3 Str/Dex, -10 Speed and -20% Stamina regen ( down from -6, -40 and -100%). The old venoms become pretty much a non-issue.

**Comment by actual-nh** [comment]
Note that the one doing localized damage will almost certainly be the hardest one to give resistance to (let alone block), given all the different digestive enzymes, etc that may be in use. (I initially was thinking that one wouldn't have any available antivenom, as being unlikely for biological/chemical warfare - but if the enzymes are hardy enough and/or can be shielded enough, they could be put in something like a hollow-tip bullet. Alexander the Great, when he invaded what is now Pakistan and India, ran into defenders smearing the toxin of the Russell's viper on their arrows and swords - see [this link](https://www.sciencedirect.com/topics/pharmacology-toxicology-and-pharmaceutical-science/snake-venom).)

File: data/json/items/comestibles/med.json [hunk 5]
```diff
@@ -1407,22 +1407,23 @@
     "type": "COMESTIBLE",
     "comestible_type": "MED",
     "name": { "str_sp": "prophylactic antivenom pill" },
-    "description": "A potent, cutting-edge antivenom in the form of some waxy white pills, designed to be taken in anticipation of venom exposure.  It was developed in top secret by the military in response to the advent of biological warfare.",
-    "weight": "1 g",
+    "description": "An experimental, cutting-edge antivenom in the form of some small amber vials, designed to be injected a few hours before venom exposure.  It was developed in top secret by the military in response to the advent of biological warfare.",
```
Comment (actual-nh) [inline data/json/items/comestibles/med.json:1410]:
```suggestion
    "description": "An experimental, cutting-edge antivenom in the form of some small amber vials, designed to be injected a few hours before venom exposure.  It was developed in top secret by the military in response to increasing biological/chemical warfare.",
```

**Pull request merged**